src/osx/cocoa/textctrl.mm

```diff
@@ -472,6 +472,27 @@ - (nullable NSUndoManager *)undoManagerForTextView:(NSTextView *)view
 }
 
 
+- (BOOL)shouldChangeTextInRange:(NSRange)affectedCharRange
+              replacementString:(NSString *)replacementString
+{
+    BOOL ret = YES;
+    wxWidgetCocoaImpl* impl = (wxWidgetCocoaImpl* ) wxWidgetImpl::FindFromWXWidget( self );
+    if ( impl )
+    {
+        wxTextCtrl *textCtrl = (wxTextCtrl *) impl->GetWXPeer();
+        if( textCtrl && textCtrl->GetTextPeer() )
+        {
+            if( textCtrl->HasFlag( wxTE_MULTILINE ) )
+            {
+                auto maxlen = textCtrl->GetMaxLen();
+                ret = [super shouldChangeTextInRange:affectedCharRange replacementString:replacementString] &&
+                      (self.string.length + (replacementString.length - affectedCharRange.length) <= maxlen);
+            }
+        }
+    }
+    return ret;
```

Comment:
This is wrong because you don't call super version for multiline controls any more. You need to do it here and return `NO` above when the change is not allowed.